### PR TITLE
Support aliased controlled gates via their base operation

### PR DESCRIPTION
**Pull request by dime10**
Small improvements to the gate decomposition strategy with the new device API.

Controlled operations defined via specialized classes (like `Toffoli` or `ControlledQubitUnitary`)
are now implemented as controlled versions of their base operation if the device supports it.
_In particular, `MultiControlledX` is no longer executed as a `QubitUnitary` with Lightning._

[sc-65228]

### CODE REVIEW

**Comment by lillian542** [comment]
Looks good! Seems like the test failures are just tests that need to be updated, but let me know when its passing and I'll give it a final look-through and approve :)

File: frontend/catalyst/device/decomposition.py [hunk 1]
```diff
@@ -47,18 +47,112 @@
...
 
 
+def check_alternative_control_support(op, capabilities):
+    """Verify that aliased controlled operations aren't supported via alternative definitions."""
+
+    if (
+        isinstance(op, qml.ControlledQubitUnitary)
+        and capabilities.native_ops.get("QubitUnitary")
+        and capabilities.native_ops.get("QubitUnitary").controllable
+    ):
```
Comment (dime10) [inline frontend/catalyst/device/decomposition.py:57]: There is probably a better way to this than long if chains but oh well 😅

**Review by lillian542 — APPROVED** [review]
Looks good to me! 🚀

File: frontend/catalyst/device/decomposition.py (hunk 1 above)
Comment (dime10) [reply frontend/catalyst/device/decomposition.py:57]: Omg is it that simple? 😅

Comment (astralcai) [reply frontend/catalyst/device/decomposition.py:57]: Yeah earlier this year I worked to make all custom controlled ops inherit from `Controlled`, which should work.

Comment (dime10) [reply frontend/catalyst/device/decomposition.py:57]: https://github.com/PennyLaneAI/catalyst/pull/1002

**Pull request merged**